### PR TITLE
Removed having and autoincrement

### PR DESCRIPTION
**Pull request by fnc12**
We have two deprecations for v1.9. It is time to remove them forever.

### CODE REVIEW

File: dev/ast/group_by.h
Comment (trueqbit) [inline dev/ast/group_by.h:1]: Need to remove [`having_t`](https://github.com/fnc12/sqlite_orm/blob/f0d5814111a3ba574fa22c308a47495ca3249468/dev/ast/group_by.h#L45)

File: dev/constraints.h
Comment (trueqbit) [inline dev/constraints.h:1]: Need to remove [`autoincrement_t`](https://github.com/fnc12/sqlite_orm/blob/f0d5814111a3ba574fa22c308a47495ca3249468/dev/constraints.h#L28)

**Review by trueqbit — CHANGES_REQUESTED** [review]
Need to remove the actual `having_t` and `autoincrement_t` structures.

File: dev/constraints.h
Comment (fnc12) [reply dev/constraints.h:1]: done

File: dev/serializing_util.h [hunk 1]
```diff
@@ -370,25 +370,10 @@ namespace sqlite_orm {
...
-                                  using constraint_type = std::decay_t<decltype(constraint)>;
-                                  constraintsStrings.push_back(serialize(constraint, context));
-                                  if(is_primary_key_v<constraint_type>) {
-                                      primaryKeyIndex = tupleIndex;
-                                  } else if(is_autoincrement_v<constraint_type>) {
-                                      autoincrementIndex = tupleIndex;
-                                  }
-                                  ++tupleIndex;
-                              });
-                if(primaryKeyIndex != -1 && autoincrementIndex != -1 && autoincrementIndex < primaryKeyIndex) {
```
Comment (fnc12) [inline dev/serializing_util.h:388]: this weird algo was the main reason why I wanted to remove this legacy

Comment (trueqbit) [reply dev/serializing_util.h:388]: Yes I see the huge change :)
So, this function can then be further simplified to only
```c++
iterate_tuple(column.constraints, [&ss, &context](auto& constraint) {
    ss << serialize(constraint, context) << ' ';
});
```

**Review by trueqbit — CHANGES_REQUESTED** [review]

File: dev/serializing_util.h (hunk 1 above)
Comment (fnc12) [reply dev/serializing_util.h:388]: fixed

**Review by trueqbit — APPROVED** [review]

**Pull request merged**